domain/user/service/service.go

```diff
@@ -19,46 +20,51 @@ import (
 type State interface {
 	// AddUser will add a new user to the database. If the user already exists
 	// an error that satisfies usererrors.AlreadyExists will be returned. If the
-	// users creator is set and does not exist then a error that satisfies
+	// users creator is set and does not exist then an error that satisfies
 	// usererrors.UserCreatorNotFound will be returned.
-	AddUser(context.Context, user.User) error
+	AddUser(ctx context.Context, uuid user.UUID, user user.User, creatorUUID user.UUID) error
 
 	// AddUserWithPasswordHash will add a new user to the database with the
 	// provided password hash and salt. If the user already exists an error that
 	// satisfies usererrors.AlreadyExists will be returned. If the users creator
-	// does not exist or has been previously removed a error that satisfies
+	// does not exist or has been previously removed an error that satisfies
 	// usererrors.UserCreatorNotFound will be returned.
-	AddUserWithPasswordHash(context.Context, user.User, string, []byte) error
+	AddUserWithPasswordHash(ctx context.Context, uuid user.UUID, usr user.User, creatorUUID user.UUID, passwordHash string, passwordSalt []byte) error
 
 	// AddUserWithActivationKey will add a new user to the database with the
 	// provided activation key. If the user already exists an error that
 	// satisfies usererrors.AlreadyExists will be returned. if the users creator
-	// does not exist or has been previously removed a error that satisfies
+	// does not exist or has been previously removed an error that satisfies
 	// usererrors.UserCreatorNotFound will be returned.
-	AddUserWithActivationKey(context.Context, user.User, []byte) error
+	AddUserWithActivationKey(ctx context.Context, uuid user.UUID, usr user.User, creatorUUID user.UUID, activationKey []byte) error
 
-	// GetUser will retrieve the user specified by name from the database where
-	// the user is active and has not been removed. If the user does not exist
+	// GetUser will retrieve the user specified by UUID from the database where
+	// the user is active. If the user does not exist
 	// or is deleted an error that satisfies usererrors.NotFound will be
```

Comment:
Think this comment needs to go. As removed users will show up in this call with the UUID.